hikari/impl/event_manager.py

```diff
@@ -367,7 +372,13 @@ async def on_guild_create(  # noqa: C901, CFQ001 - Function too complex and too
         # payload if presence intents are also declared, so if this isn't the case then we also want
         # to chunk small guilds.
         if (
-            self._auto_chunk_members
+            (
+                self._auto_chunk_members is True
```

Comment:
This `auto_chunk_members` change seems to just over-complicate it imo.
Why can't the `self._cache_enabled_for(config.CacheComponents.MEMBERS)` bit just have a check added on to make it 

```py
(
    self._cache and self._cache_enabled_for(config.CacheComponents.MEMBERS)
    and not self._cache.settings.only_my_member
)
```

and avoid changing the type of `auto_chunk_members` all together